File: 206-ReverseLinkedList/206-ReverseLinkedList.c
// Last Updated: 6/22/2026, 12:42:37 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 typedef struct ListNode node;
struct ListNode* reverseList(struct ListNode* head) {
    node *pn=NULL;
    node *cn=head;
    node *nn=head;
    while(cn!=NULL){
        nn=nn->next;
        cn->next=pn;
        pn=cn;
        cn=nn;  
    }
    return pn;
    
}

File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last Updated: 6/22/2026, 12:42:32 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cn=head
        pn=None
        while cn:
            nn=cn.next
            cn.next=pn
            pn=cn
            cn=nn
        return pn

